docs/DeveloperGuide.md

```diff
@@ -1,34 +1,565 @@
+
+> :bulb: The application only accepts `.txt` files.
+
+#### 4.1.1 Current Implementation
+
+The `estimate` feature is facilitated by `EstimateCommand`, which extends from the abstract class `Command`. `EstimateCommand` is instantiated by `EstimateParser`, and it requires two parameters:
+*   `ScriptFileName` (mandatory).
+*   `wordsPerHour` (optional), if not specified, the values 400, 500, and 600 words per hour (average translator's speed) will be used to generate 3 estimation timings.
+<br/>
+
+Given below is an example usage scenario showing how the `estimate` command behaves at each step.
+
+**Step 1:** User executes `estimate script.txt -wph 300`, `Main` calls `Parser#getCommand()` and passes the command to it.
+
+**Step 2:** `Parser` extracts “estimate” from the command, and according to the command type, it calls `EstimateParser#parse()` and passes the command description, "script.txt -wph 300", to it.
+
+**Step 3:** `EstimateParser` proceed to parse the command description, extract, and validate the inputs "script.txt" and "300". Once validated and no exception was thrown, it creates and initialises `EstimateCommand` with the inputs and returns this object to `Parser`, which then returns it to `Main`.
+
+**Step 4:** `EstimateParser` is terminated and `Main` calls `EstimateCommand#execute()` with `animeData`, `storageManager`, and `user` to begin estimating the time needed.
+
+![Estimate Command After Step 4 Diagram](images/EstimateCommand-After-Step-4.png) <br/>
+*Figure 9: Estimate Command After Step 4*
+
+<br/>
+
+**Step 5:** `EstimateCommand` first calls `User#getActiveWorkspace()` to initialise a `Workspace` object, `activeWorkspace`. It then calls `StorageManager#loadScript()` with `scriptFileName` and `activeWorkspace.getName()` to initialise `fileContent`, with the content of `scriptFileName` located in the workspace folder named `activeWorkspace.getName()`.
+
+**Step 6:** `EstimateCommand` calculates the estimated time using `fileContent` and `wordsPerHour`. It then calls `EstimateCommand#timeNeededToString()` with the estimated time to convert the timing into a human-readable format and return the result to `Main` for it to be printed to the user via `Ui`.
+
+**Step 7:** `EstimateCommand` is terminated.
+
+![Estimate Command Final State Diagram](images/EstimateCommand-Final-State.png) <br/>
+*Figure 10: Estimate Command Final State*
+
+<br/>
+
+The sequence diagram presented below depicts the interaction between the components for running the `estimate` command, provided that the user has entered a valid command.
+
+<br/>
+
+![Estimate Command Sequence Diagram](images/EstimateCommand-Sequence-Diagram.png) <br/>
+*Figure 11. Sequence Diagram for EstimateCommand*
+
+<br/>
+
+#### 4.1.2 Design Consideration
+
+This section shows some design considerations taken when implementing the estimate feature.
+
+<br/>
+
+##### Aspect: When should the program read the script file
+
```

Comment:
![Capture](https://user-images.githubusercontent.com/60357492/97390678-5f142f80-1918-11eb-9e3e-4ae6bc346e76.PNG)
While i feel the explanations here are well done, It could be too wordy and overly complicated. It should be easier to understand with a general flow diagram to justify your design choices.